PublicSDKSwiftAPITester/PurchasesAPI.swift

```diff
@@ -0,0 +1,166 @@
+
+    Purchases.configure(withAPIKey: "", appUserID: "", observerMode: false)
+
+    Purchases.configure(withAPIKey: "", appUserID: "", observerMode: true, userDefaults: nil)
+    Purchases.configure(withAPIKey: "", appUserID: "", observerMode: true, userDefaults: UserDefaults())
+
+    // static methods
+    let logHandler: (Purchases.LogLevel, String) -> Void = { _, _ in }
+    Purchases.setLogHandler(logHandler)
+
+    let canI: Bool = Purchases.canMakePayments()
+    let version = Purchases.frameworkVersion
+
+    Purchases.addAttributionData([AnyHashable: Any](), from: RCAttributionNetwork.adjust, forNetworkUserId: "")
+    Purchases.addAttributionData([AnyHashable: Any](), from: RCAttributionNetwork.adjust, forNetworkUserId: nil)
+
+    let automaticAppleSearchAdsAttributionCollection: Bool = Purchases.automaticAppleSearchAdsAttributionCollection
+    let debugLogsEnabled: Bool = Purchases.debugLogsEnabled
+    let logLevel = Purchases.logLevel
+    let proxyUrl: URL? = Purchases.proxyURL
+    let forceUniversalAppStore: Bool = Purchases.forceUniversalAppStore
+    let simulatesAskToBuyInSandbox: Bool = Purchases.simulatesAskToBuyInSandbox
+    let sharedPurchases: Purchases = Purchases.shared
+    let isConfigured: Bool = Purchases.isConfigured
+    let finishTransactions: Bool = purch.finishTransactions
+    let delegate: PurchasesDelegate? = purch.delegate
+    let appUserID: String = purch.appUserID
+    let isAnonymous: Bool = purch.isAnonymous
+
+    print(canI, version, automaticAppleSearchAdsAttributionCollection, debugLogsEnabled, logLevel, proxyUrl!,
+          forceUniversalAppStore, simulatesAskToBuyInSandbox, sharedPurchases, isConfigured, finishTransactions,
+          delegate!, appUserID, isAnonymous)
+
+    checkPurchasesSubscriberAttributesAPI(purchases: purch)
+    checkPurchasesPurchasingAPI(purchases: purch)
+
+    let piComplete: Purchases.ReceivePurchaserInfoBlock = { _, _ in }
+    // identity
+    purch.createAlias("", piComplete)
+    purch.identify("", piComplete)
+    purch.reset(piComplete)
+    purch.logOut(piComplete)
+
+    let loginComplete: (Purchases.PurchaserInfo?, Bool, Error?) -> Void = { _, _, _ in }
+    purch.logIn("", loginComplete)
+}
+
+func checkPurchasesEnums() {
+    var type = Purchases.PeriodType.normal
+    type = Purchases.PeriodType.intro
+    type = Purchases.PeriodType.trial
+
+    var oType: RCPurchaseOwnershipType = RCPurchaseOwnershipType.purchased
+    oType = RCPurchaseOwnershipType.familyShared
+    oType = RCPurchaseOwnershipType.unknown
+
+    var logLevel: Purchases.LogLevel = Purchases.LogLevel.info
+    logLevel = Purchases.LogLevel.warn
+    logLevel = Purchases.LogLevel.debug
+    logLevel = Purchases.LogLevel.error
+
+    print(type, oType, logLevel)
+}
+
+func checkConstants() {
+    // were these never available from swift?
+//    let versionNum: Double = PurchasesVersionNumber
+//    let versionString: String = PurchasesVersionString
```

Comment:
probably not, they're just auto-generated when you create an obj-c framework, but I don't think they have any use for customers